data_validation/partition_builder.py

```diff
@@ -40,17 +41,38 @@ def _get_arg_config_dir(self) -> str:
 
         return self.args.config_dir
 
-    def _get_yaml_from_config(self, config_manager: ConfigManager) -> Dict:
-        """Return dict objects formatted for yaml validations.
+    def _add_filters_get_yaml(
+        self,
+        config_manager: ConfigManager,
+        source_filters: List[str],
+        target_filters: List[str],
+    ) -> Dict:
+        """Return dict objects formatted for yaml validations. Qualify the configurations by adding the source and
+            target filters which validate individual partitions. Each filter pair will result in a DVT execution. Running multiple validations
+            sequentially spreads the startup and shutdown overhead of containers accross multiple partitions.
+            Parallelism can be achieved by running multiple yaml files in parallel.
```

Comment:
Let's keep the docstring for each function limited to the input/output + goal of the function - otherwise it gets hard to understand. We should avoid documenting best practices/explaining DVT usage here. End users won't be reading this, it should be for developers only.

We need to update the 'Args' in this docstring with source/target_filters. There we can explain which type of filter this is adding to the config (existing or new).